src/ensembl/src/shared/components/help-popup/useHelpArticle.ts

```diff
@@ -0,0 +1,129 @@
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState, useEffect } from 'react';
+
+import config from 'config';
+
+import useApiService from 'src/shared/hooks/useApiService';
+
+import {
+  HelpArticle,
+  HelpVideo,
+  RelatedArticle,
+  SlugReference,
+  PathReference
+} from './types';
+
+const getQuery = (params: { reference: SlugReference | PathReference }) => {
+  if ('slug' in params.reference) {
+    return `slug=${params.reference.slug}`;
+  } else {
+    return `path=${encodeURIComponent(params.reference.path)}`;
+  }
+};
+
+type Reference = SlugReference | PathReference;
+
+export type ArticleReference = {
+  type: 'article';
+  reference: Reference;
+};
+
+export type VideoReference = {
+  type: 'video';
+  youtube_id: string;
+};
+
+export type RelatedItems = {
+  articles: RelatedArticle[];
+  videos: HelpVideo[];
+};
+
+export type CurrentArticle = HelpArticle & { type: 'article' };
+export type CurrentVideo = HelpVideo & { type: 'video' };
+export type CurrentItem = CurrentArticle | CurrentVideo;
+
+export const emptyRelatedItems: RelatedItems = {
+  articles: [],
+  videos: []
+};
+
+const useHelpArticle = (reference: ArticleReference | VideoReference) => {
+  const [currentHelpItem, setCurrentHelpItem] = useState<CurrentItem | null>(
+    null
+  );
+  const [relatedItems, setRelatedItems] = useState<RelatedItems | null>(null);
+
+  const query = reference.type === 'article' ? getQuery(reference) : null;
+  const { helpApiHost } = config;
+  const url = query ? `${helpApiHost}/api/article?${query}` : '';
+
+  const { data: article, loadingState } = useApiService<HelpArticle>({
+    endpoint: url,
+    skip: !url
+  });
+
+  useEffect(() => {
+    if (
+      article &&
+      reference.type === 'article' &&
+      (('slug' in reference.reference &&
+        reference.reference.slug === article.slug) ||
+        ('path' in reference.reference &&
+          reference.reference.path === article.path))
+    ) {
+      setCurrentHelpItem({
+        ...article,
+        type: 'article'
+      });
+    } else if (relatedItems && reference.type === 'video') {
+      const video = relatedItems.videos.find(
+        (video) => video.youtube_id === reference.youtube_id
+      ) as HelpVideo;
+      setCurrentHelpItem({ ...video, type: 'video' });
+    }
+  }, [article?.path, reference.type]);
+
+  useEffect(() => {
+    if (!article || relatedItems) {
+      return;
+    }
+    // keep track only for the article that was fetched initially
+    setRelatedItems(prepareRelatedItems(article));
+  });
+
+  return {
+    loadingState,
+    relatedItems,
+    currentHelpItem
```

Comment:
Would it be better to consistently name these vars as `HelpItems` everywhere?

Reply:
Not sure what you mean. I used the word _article_ to refer to a text document, _video_ to refer to a video, and _item_ to refer to things that can be either of those two. Where would you make the changes?

Reply:
Sorry, I meant renaming `relatedItems` to `relatedHelpItems` to keep it consistent with `currentHelpItem`.

Reply:
Updated the name.